src/main/java/com/exactpro/th2/common/schema/grpc/router/impl/DefaultGrpcRouter.java

```diff
@@ -61,7 +65,28 @@
      * @throws ClassNotFoundException if matching the service class to protobuf stub has failed
      */
     public <T> T getService(@NotNull Class<T> cls) throws ClassNotFoundException {
-        return getProxyService(cls);
+        Objects.requireNonNull(cls, "Services class can not be null");
+
+        TH2Impl implementationAnnotation = cls.getAnnotation(TH2Impl.class);
+
+        if (implementationAnnotation == null || implementationAnnotation.value() == null) {
```

Comment:
I don't think that the TH2Impl annotation with the `null` value is a normal case. Probably, we should throw an exception in this case.
If the annotation is not present at all it is ok. But if it presents it must have a non-null value